### PR TITLE
Feature/block username emails

### PR DESCRIPTION
**Pull request by B-Griffinn**
# Codenames & Emails 

We do not want Codenames (usernames) to be emails to prevent web scrappers and people from getting our users personal emails.
We want users passwords to follow the front end rules of length, a capital letter and a number as well.

## Auth Token

Send a jwt token to the FE when a user is activated so the user can signed in automatically opposed to manually logging in after activation.

### CODE REVIEW

File: api/email/emailModel.js [hunk 1]
```diff
@@ -113,4 +114,27 @@ async function saveResetCode(uid, code)
     {
         console.log(ex); return -1;
     }
+}
+
+// Regex checking if the username only contains letters and numbers
+async function userNamePattern(username)
```
Comment (lgv-0) [inline api/email/emailModel.js]: [Test](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test) shouldn't need an async

File: api/email/emailModel.js [hunk 2]
```diff
@@ -113,4 +114,27 @@ async function saveResetCode(uid, code)
...
+            else 
+            {
+                return false
+            }
+        }
+        catch (error) 
+        {
+            console.log(error)   
+        }
+    }
```
Comment (lgv-0) [inline api/email/emailModel.js]: Is there an extra bracket encasing here?

File: api/email/emailModel.js [hunk 3]
```diff
@@ -113,4 +114,27 @@ async function saveResetCode(uid, code)
...
         console.log(ex); return -1;
     }
+}
+
+// Regex checking if the username only contains letters and numbers
+async function userNamePattern(username) 
+{
+    {
+        const codenamePattern = /^[A-Za-z0-9]*$/;
+        try
```
Comment (lgv-0) [inline api/email/emailModel.js]: Shouldn't need try/catch, test shouldn't have any uncaught throws - we should probably add some type validation to the forms though thinking about it now!

**Review by lgv-0 — CHANGES_REQUESTED** [review]

File: api/email/emailModel.js (hunk 2 above)
Comment (B-Griffinn) [reply api/email/emailModel.js:139]: @lgv-0 if a try catch is not needed then we should be good once your type validation aligns with my vector

**Review by lgv-0 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**